src/RageTimer.cpp

```diff
@@ -51,14 +51,21 @@ static std::uint64_t GetTime( bool /* bAccurate */ )
 #endif
 }
 
-float RageTimer::GetTimeSinceStart( bool bAccurate )
+/* In the past, this was float instead of double to maintain compatibility
+with legacy hardware that can't use the double type. On modern hardware,
+performance of double is good or better than mixing 32 bit and 64 bit wide
+types, and maintaining precision is directly related to the stability of clock
+sync during gameplay. Too much error here will manifest as a drastic shift
+in the game's sync (enough to push a Fantastic into the Great window or worse).
+Incorrect math here will manifest as a consistent sync offset in game, similar
+to the effect of changing the global offset. */
+double RageTimer::GetTimeSinceStart(bool bAccurate)
 {
-	std::uint64_t usecs = GetTime( bAccurate );
+	std::uint64_t usecs = GetTime(bAccurate);
 	usecs -= g_iStartTime;
-	/* Avoid using doubles for hardware that doesn't support them.
-	 * This is writing usecs = high*2^32 + low and doing
+	/* This is writing usecs = high*2^32 + low and doing
 	 * usecs/10^6 = high * (2^32/10^6) + low/10^6. */
-	return std::uint32_t(usecs>>32) * 4294.967296f + std::uint32_t(usecs)/1000000.f;
+	return double(usecs >> 32) * 4294.967296 + double(usecs) / 1000000.0;
```

Comment:
We shouldn't need the explicit `double` conversions here.

We can just do `return (usecs >> 32) * 4294.967296 + (usecs) / 1000000.0;` (since the intermediates are doubles because of the double math, and `usecs` retains `uint64_t` from the variable type.

Reply:
Sounds good.